tests/interfaces_t/test_siteadapter.py

```diff
@@ -0,0 +1,86 @@
+from tardis.interfaces.siteadapter import SiteAdapter
+from tardis.utilities.attributedict import AttributeDict
+
+from ..utilities.utilities import run_async
+
+from unittest import TestCase
+from unittest.mock import patch
+
+
+class TestSiteAdapter(TestCase):
+    @patch.multiple(SiteAdapter, __abstractmethods__=set())
+    def setUp(self) -> None:
+        self.site_adapter = SiteAdapter()
+
+    def test_configuration(self):
+        with self.assertRaises(AttributeError):
+            self.site_adapter.configuration
```

Comment:
Please consider also adding tests for cases where no ``AttributeError`` is raised (if those are not already implemented).

Reply:
This is already covered in the tests of sub-classes of the `SiteAdapter`. However, I can add it to have a well-defined unittest for the interface class.